R/duckreg.R

```diff
@@ -2,220 +2,731 @@
-#' connection will be created automatically and closed before the function
-#' exits. Note that a persistent (disk-backed) database connection is
-#' required for larger-than-RAM datasets in order to take advantage of DuckDB's
-#' streaming functionality.
-#' @param table,data,path Mututally exclusive arguments for specifying the data
-#' table (object) to be queried. In order of precedence:
-#' - `table`: Character string giving the name of the data table in an
-#' existing (open) DuckDB connection.
-#' - `data`: R dataframe that can be copied over to `conn` as a temporary
-#' table for querying via the DuckDB query engine. Ignored if `table` is
-#' provided.
-#' - `path`: Character string giving a path to the data file(s) on disk, which
-#' will be read into `conn`. Internally, this string is passed to the `FROM`
-#' query statement, so could (should) include file globbing for
-#' Hive-partitioned datasets, e.g. `"mydata/**/.*parquet"`. For more precision,
-#' however, it is recommended to pass the desired DuckDB reader function as
-#' part of this string, e.g. `"read_parquet('mydata/**/*.parquet')"`;
-#' note the use of single quotes.
-#' Ignored if either `table` or `data` is provided. 
-#' @param vcov Character string denoting the desired type of variance-
-#' covariance correction / standard errors. At present, only "hc1"
-#' (heteroskedasticity-consistent) are supported, which is also thus
-#' the default.
-#' @param query_only Logical indicating whether only the underlying compression
-#'   SQL query should be returned (i.e., no computation will be performed).
-#'   Default is `FALSE`.
-#' @param data_only Logical indicating whether only the compressed dataset
-#'   should be returned (i.e., no regression is run). Default is `FALSE`.
-#' 
-#' @return A list of class "duckreg" containing various slots, including a table
-#' of coefficients (which the associated print method will display).
-#' @references
-#' Wong, J., Forsell, E., Lewis, R., Mao, T., & Wardrop, M. (2021).
-#' \cite{You Only Compress Once: Optimal Data Compression for Estimating Linear Models.} 
-#' arXiv preprint arXiv:2102.11297.
-#' Available: https://doi.org/10.48550/arXiv.2102.11297
-#' 
+#' Internal strategies:
+#' 1. group: GROUP BY regressors (+ fixed effects) -> frequency-weighted rows -> WLS.
+#' 2. moments: Global sufficient statistics (X'X, X'y) (no fixed effects).
+#' 3. moments_fe: One or two-way fixed-effect within (demean/double-demean) + cross-moments.
+#'
+#' Auto heuristic:
+#' - If no FEs and (any regressor appears continuous OR estimated compression ratio > compression_threshold) => moments.
+#' - If 1-2 FEs and estimated compression ratio high (> max(0.6, compression_threshold)) => moments_fe.
+#' - Else group.
+#'
+#' Notes:
+#' - Robust HC1: group path computes HC1; moments/moments_fe apply a df correction to classical variance.
+#'
+#' @param fml Formula. Use pipe for FEs: y ~ x1 + x2 | fe1 + fe2
```

Comment:
Hmmmm. Is there a reason for rewriting my old param entries? Some of these are a little terse / imprecise ffor my liking.